### ISSUE TITLE
require-sri does not recognize backslash-spelled cross-origin URLs

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#5239]
## Summary

`html-validate`'s `require-sri` rule does not flag a cross-origin subresource whose URL is spelled with backslashes, though browsers resolve it to the same cross-origin URL as the forward-slash form.

## Reproduction

Against this repo's pinned `html-validate` 11.10.0 and committed `.htmlvalidate.json`:

```console
$ printf '%s\n' '<!doctype html><html lang="en"><head><title>x</title><link rel="stylesheet" href="https:\\cdn.example/x.css"></head><body></body></html>' \
  | ./node_modules/.bin/html-validate --config .htmlvalidate.json --stdin ; echo "exit=$?"
exit=0

$ printf '%s\n' '<!doctype html><html lang="en"><head><title>x</title><link rel="stylesheet" href="https://cdn.example/x.css"></head><body></body></html>' \
  | ./node_modules/.bin/html-validate --config .htmlvalidate.json --stdin ; echo "exit=$?"
  1:55  error  SRI "integrity" attribute is required on <link> element  require-sri
exit=1
```

Both are the same resource to a browser:

```console
$ node -e 'console.log(new URL(String.raw`https:\\cdn.example/x.css`).href)'
https://cdn.example/x.css
```

Per the [WHATWG URL standard](https://url.spec.whatwg.org/#special-scheme), `\` is normalized to `/` after a special scheme, so the two spellings are equivalent. The repo's bespoke markup gate in `test/toolchain.test.ts` does not catch it either.

## Severity

Low as a live risk: no such spelling exists in the tree, and `index.html` is hand-written rather than attacker-controlled. This is gate coverage, not exposure.

## Proposed handling

Report upstream to `html-validate` rather than hand-writing a URL normalizer here. Per principles 1 and 6 in `docs/design/web-frontend-security.md`, a preset's incomplete coverage is an upstream fix; a local substitute would recreate exactly the enumeration problem that motivated adopting ecosystem linters in the first place.

Recorded as a known gap in #5237. Found by adversarial review of that PR.

**Comment by richlander** [comment]
**Correction to the original report, and a second defect.**

Round 2 review of #5237 showed that this issue as filed was partly wrong and partly understated.

**Wrong:** I wrote that "the repo's bespoke markup gate in `test/toolchain.test.ts` does not catch it either." That is false for `<script>`. The gate normalizes backslashes deliberately, with a comment saying why:

```ts
// `//` is not the only spelling of an authority: the URL parser treats a backslash as a
// slash for special schemes, so `/\host/x.js`, `\\host/x.js` and `\/host/x.js` all
// resolve to `https://host/x.js`. Normalizing first is what makes one comparison cover
// every spelling, and getting this wrong would skip the `integrity` requirement below.
const normalized = value.replaceAll("\\", "/");
```

The round 1 probe that returned `[]` used a `<link rel="stylesheet">`, and the gate's integrity requirement is inside an `if (tag.element === "script")` branch. So the gate covers remote scripts including backslash spellings, and does not cover `<link>` at all. I generalized from one element to all elements.

**Understated:** `require-sri` does not validate the digest either, only that the attribute exists. Against the pinned 11.10.0 and committed config:

```console
$ printf '... <link rel="stylesheet" href="https://cdn.example/x.css" integrity="not-a-digest"> ...' \
  | ./node_modules/.bin/html-validate --config .htmlvalidate.json --stdin ; echo "exit=$?"
exit=0
```

An `integrity` value that pins nothing satisfies the rule. The repo's own gate has a `pinsItsBytes` check for exactly this reason — again, scripts only.

**Restated scope:** the uncovered case is a cross-origin `<link>` whose `integrity` is absent or malformed. Both halves — spelling coverage and digest validity — are `html-validate` behavior and belong upstream; the local gate already demonstrates both properties for scripts.
